src/js/components/Form/Form.js

```diff
@@ -165,16 +174,26 @@ const Form = forwardRef(
     ref,
   ) => {
     const { format } = useContext(MessageContext);
-
     const [valueState, setValueState] = useState(valueProp || defaultValue);
     const value = useMemo(
       () => valueProp || valueState,
       [valueProp, valueState],
     );
     const [touched, setTouched] = useState(defaultTouched);
-    const [validationResults, setValidationResults] = useState(
-      defaultValidationResults,
-    );
+    const [validationResults, setValidationResults] = useState({
+      errors: errorsProp,
+      infos: infosProp,
+    });
+    const validationResultsRef = useRef({});
+    // Simulated onMount state. Consider Form to be mounted once it has
```

Comment:
Open to alternate ideas here. Played with a number of approaches, but landed here.

Reply:
I am okay with this approach

Reply:
hmm... My initial reaction is that I'm wondering if there's a more semantic way of attacking this, rather than relying on the number of renderings. I'll see if this sticks after reviewing more.

Reply:
If I'm understanding the need for this, it's that Form renders something like a TextInput and the TextInput calls `useFormInput` with the controlled `value` but Form isn't able to put that in its `valueState` right away and instead waits for a `useEffect` to put it in later, after the first render. Have I got that right?

Reply:
This approach seems vulnerable to a Form being rendered in multiple steps. For instance, the caller renders the form with the data it has from an API request but then sees that one of the values requires another API request to get the second chunk of data. The user is shown the initial form and a Spinner as it waits to load the rest of the form. This is a bit contrived but I'm wondering if the basic pattern could still occur.

Reply:
I'm wondering if we can queue up the initial controlled values inside Form in a ref and let the Form handle both `setValueState` and running mount time validations in an effect of its own, instead of relying on an effect inside `useFormInput`? So:
- TextInput calls `useFormInput` with the controlled `value`
- this `useFormInput` adds the value to a ref inside Form, call it `inputValuesRef`.
- Form has an effect that checks if there are any values in `inputValuesRef`, adds them to `valueState`, runs any mount time validations, and clears `inputValuesRef`.
I'm not 100% sure, just contemplating. Thoughts?